### DISCUSSION TITLE
turbo run dev with dependent commands

### DISCUSSION THREAD

**Discussion by arkmech** [vercel/turborepo#5850] [Help] [answered]
`turbo run dev --parallel --filter=packageA --filter=packageB --filter=packageC`

**Question**
Is there a way to specify each package's dev command dependent on another?

**Info**
- packageC and packageB are are `tsup` builds.
- packageA depends on packageB and packageC.
- packageB depends on packageC.

There is some delay generating TS declaration files with `tsup`.

1. The delay is large enough that if packageB is looking for packageC's type declaration files, but doesn't find them in time. And it errors (Declaration File not found).
2. This causes packageB's `tsup` build fail which causes to fail generating a `.css` file.
3. Which then in turn causes packageA (Next.js) app that consumes the `.css` file to fail because its looking for a `.css` file that doesn't exist in packageB.

**Comment by mehulkar** [comment, marked as answer]
This is not possible at the moment, but you're looking for some version of https://github.com/vercel/turbo/issues/986

**Reply by codenoobforreal** [reply]
My question is similar.
I have a express app and a react app.The react app uses express api to get the initial data for index page.
So what I expected is I can write some configurations below then exec `turbo run dev` to run express app first.
```js
"dev": {
     "first":"express",
 },
```
